### PR TITLE
Feature enhance help

### PR DESCRIPTION
**Pull request by mosheBelostotsky**

### CODE REVIEW

File: src/0_bootstrap.coffee [hunk 1]
```diff
@@ -33,7 +33,10 @@ insight = new Insight(
 
 insight.optOut = process.env.HUBOT_HE_OPT_OUT || false
 insight.track 'HE', 'start'
-reserverd_apps = [ 'info', 'help' ]
+help_words = [ 'info', 'menu', 'home', '\\?', 'support', 'help']
+# contatination of help
+reserverd_apps = help_words.concat([])
+help_words.push('enterprise')
```
Comment (eitzhaki) [inline src/0_bootstrap.coffee]: please add a comment "why here"

File: src/0_bootstrap.coffee [hunk 2]
```diff
@@ -274,60 +279,90 @@ module.exports = (robot) ->
...
+  # process help and show message (accept hubot middleware)
+  #  msg: hubot message object (context.response)
+  processHelp = (msg) ->
+    # build help regex from help workds array
+    help_re = robot.respondPattern(new RegExp("("+help_words.join('|')+")"+
+      "[ ]?(\\w+)?[ ]?(\\w+)?[ ]?(\\w+)?", 'i'))
+    if (msg.match = msg.message.text.match(help_re))
+      help = robot.e.show_help(msg.match[2], msg.match[3], msg.match[4])
+      robot.logger.debug help
+      msg.message.text = msg.message.text.replace(msg.match[1], 'help')
```
Comment (eitzhaki) [inline src/0_bootstrap.coffee]: Add comment for hubot does not support for example "?"

File: src/0_bootstrap.coffee (hunk 1 above)
Comment (mosheBelostotsky) [inline src/0_bootstrap.coffee]: DONe

File: src/0_bootstrap.coffee (hunk 2 above)
Comment (mosheBelostotsky) [inline src/0_bootstrap.coffee]: DONE

**Pull request merged**